pkg/operator/ceph/cluster/rbd/mirror.go

```diff
@@ -70,22 +82,25 @@ func New(context *clusterd.Context, namespace, rookVersion string, cephVersion c
 func (m *Mirroring) Start() error {
 	logger.Infof("configure rbd mirroring with %d workers", m.spec.Workers)
 
-	access := []string{"mon", "profile rbd-mirror", "osd", "profile rbd"}
-
 	for i := 0; i < m.spec.Workers; i++ {
-		daemonName := k8sutil.IndexToName(i)
-		username := fullDaemonName(daemonName)
-		resourceName := fmt.Sprintf("%s-%s", appName, daemonName)
-		cfg := spec.KeyringConfig{Namespace: m.Namespace, ResourceName: resourceName, DaemonName: daemonName, OwnerRef: m.ownerRef, Username: username, Access: access}
-		if err := spec.CreateKeyring(m.context, cfg); err != nil {
-			return fmt.Errorf("failed to create %s keyring. %+v", resourceName, err)
+		daemonID := k8sutil.IndexToName(i)
+		resourceName := fmt.Sprintf("%s-%s", appName, daemonID)
+		daemonConf := &daemonConfig{
+			DaemonID:     daemonID,
+			ResourceName: resourceName,
+			DataPathMap:  config.NewDatalessDaemonDataPathMap(),
+		}
+
+		if err := m.generateKeyring(daemonConf); err != nil {
+			return fmt.Errorf("failed to generate keyring for %s. %+v", resourceName, err)
 		}
 
 		// Start the deployment
-		deployment := m.makeDeployment(resourceName, daemonName)
+		deployment := m.makeDeployment(daemonConf)
 		if _, err := m.context.Clientset.Apps().Deployments(m.Namespace).Create(deployment); err != nil {
 			if !errors.IsAlreadyExists(err) {
 				return fmt.Errorf("failed to create %s deployment. %+v", resourceName, err)
+				/* TODO: update rbd mirrors if already exist */
```

Comment:
This will be a separate PR? Definitely would be good to take care of this or else we won't get upgrades.

Reply:
Ah I see it's #2804